agent/utils/ephemeral_ports.go

```diff
@@ -89,20 +90,26 @@ func (pt *safePortTracker) GetLastAssignedHostPort() int {
 	return pt.lastAssignedHostPort
 }
 
-var dynamicHostPortRange = getDynamicHostPortRange
+var dynamicHostPortRange = GetDynamicHostPortRange
 var tracker safePortTracker
 
 // GetHostPortRange gets N contiguous host ports from the ephemeral host port range defined on the host.
-func GetHostPortRange(numberOfPorts int, protocol string) (string, error) {
+func GetHostPortRange(numberOfPorts int, protocol string, portRange string) (string, error) {
 	portLock.Lock()
 	defer portLock.Unlock()
 
 	// get ephemeral port range, either default or if custom-defined
-	startHostPortRange, endHostPortRange, err := dynamicHostPortRange()
+	startHostPortRange, endHostPortRange, err := nat.ParsePortRangeToInt(portRange) //dynamicHostPortRange()
```

Comment:
we need to validate this when we parse the config and load it, not here. 

so in the **load config workflow**, we would – 
check if `ECS_DYNAMIC_HOST_PORT_RANGE` is set. If it is set, validate using this `nat.ParsePortRangeToInt()`. if it is valid, set the value of `cfg.DynamicHostPortRange` as provided, else default to whatever`utils.GetDynamicHostPortRange` method gives you.


all of this should be in `config/parse.go`